app/models/mhv_account.rb

```diff
@@ -93,14 +92,44 @@ def user
     @user ||= User.find(user_uuid)
   end
 
-  # TODO: remove this in future migration
-  def mhv_correlation_id
-    user.mhv_correlation_id
+  def identity_proofed?
+    user.loa3?
+  end
+
+  def va_patient?
+    user.va_patient?
+  end
+
+  def ssn_mismatch?
+    user.ssn_mismatch?
+  end
+
+  def multiple_active_mhv_ids?
+    return false if previously_upgraded? || previously_registered?
+    user.va_profile.active_mhv_ids.size > 1
+  end
+
+  # for now lets not handle this
+  def state_eligible?
+    true
+  end
+
+  # fot now lets not handle this
+  def country_eligible?
+    true
+  end
+
+  def previously_upgraded?
+    exists? && unknown? && upgraded_at?
```

Comment:
This reads kind of weird with checking if the state is `unknown`. Is it worth (for the sake of readability) adding another state called `eligible` to be explicit and having that be the terminal state for `check_eligibility`? In a sense, we kind of know at least the user passed eligibility checks, so the state isn't entirely unknown.

Reply:
We could keep `unknown` as the initial state. Another thing we could get out of that is that if the user remains `unknown` in the end, perhaps we have entered some unhandled state that we should be logging.